### PR TITLE
Add separate URL for sign up check email page

### PR DESCRIPTION
**Pull request by C-gyorfi**
## Context

The `sign-in/check-your-email` page: needs separate URL and post redirect get

## Changes proposed in this pull request

This PR adds separate URL for the following paths:
- `/candidate/sign-up/check-email`
- `/candidate/sign-in/check-email`
- `/candidate/not_eligible`

## Guidance to review

Try to sign-in, sign-up and you should be able to see the correct URL. 

## Link to Trello card

 https://trello.com/c/JJm9jjSo/430-sign-up-check-email-page-needs-separate-url-and-post-redirect-get

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -21,12 +21,15 @@
 
     get '/eligibility' => 'start_page#eligibility', as: :eligibility
     post '/eligibility' => 'start_page#determine_eligibility'
+    get '/not_eligible', to: 'start_page#not_eligible', as: :not_eligible
 
     get '/sign-up', to: 'sign_up#new', as: :sign_up
     post '/sign-up', to: 'sign_up#create'
+    get '/check-email/sign-up', to: 'sign_in#check_your_email', as: :check_email_sign_up
```
Comment (chubberlisk) [inline config/routes.rb]: To me this makes more sense, sign-up is the "top-level" thing and "check-email" is a part of sign-up.

```suggestion
    get '/sign-up/check-email', to: 'sign_in#check_your_email', as: :sign_up_check_email
```

File: config/routes.rb [hunk 2]
```diff
@@ -21,12 +21,15 @@
...
 
     get '/sign-in', to: 'sign_in#new', as: :sign_in
     post '/sign-in', to: 'sign_in#create'
+    get '/check-email/sign-in', to: 'sign_in#check_your_email', as: :check_email_sign_in
```
Comment (chubberlisk) [inline config/routes.rb]: Same reasoning here.

```suggestion
    get '/sign-in/check-email', to: 'sign_in#check_your_email', as: :sign_in_check_email
```

File: config/routes.rb [hunk 3]
```diff
@@ -21,12 +21,15 @@
 
     get '/eligibility' => 'start_page#eligibility', as: :eligibility
     post '/eligibility' => 'start_page#determine_eligibility'
+    get '/not_eligible', to: 'start_page#not_eligible', as: :not_eligible
```
Comment (chubberlisk) [inline config/routes.rb]:
```suggestion
    get '/not-eligible', to: 'start_page#not_eligible', as: :not_eligible
```

File: config/routes.rb (hunk 1 above)
Comment (C-gyorfi) [reply config/routes.rb]: I was following the pattern from the prototype, but probably makes more sense to use `/sign-up/check-email` 👍

File: config/routes.rb (hunk 3 above)
Comment (C-gyorfi) [reply config/routes.rb]: Resolved in dafafa3

File: config/routes.rb (hunk 2 above)
Comment (C-gyorfi) [reply config/routes.rb]: Resolved in dafafa3

File: config/routes.rb (hunk 1 above)
Comment (C-gyorfi) [reply config/routes.rb]: Resolved in dafafa3

**Pull request merged**